src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_utils.hpp

```diff
@@ -42,14 +43,19 @@ size_t compute_vnni_factor(const ov::element::Type& precision);
 size_t get_elems_in_vec(const ov::element::Type& precision);
 
 namespace repacking {
+/// \brief  Computes inner N block size used by OneDNN implementation. Depends on tensor precision
+size_t compute_inner_n_block(const ov::element::Type& precision);
 /**
  * @brief Computes leading dimension (LDB) which must be used in brgemm and brgemm_copy_b emitters
  * @param n_block N block size shared between BrgemmCPU and BrgemmCopyB node
  * @param precision tensor precision
  */
-size_t compute_out_leading_dim(const size_t n_block, const ov::element::Type& precision);
-/// \brief  Computes inner N block size used by OneDNN implementation. Depends on tensor precision
-size_t compute_inner_n_block(const ov::element::Type& precision);
+template<typename T, typename = typename std::enable_if<(std::is_same<T, size_t>::value || std::is_same<T, int64_t>::value), bool>::type>
+T compute_out_leading_dim(T n_block, const ov::element::Type& precision) {
+    return snippets::utils::is_dynamic_value<T>(n_block) ?
+           n_block :
+           std::max(n_block, static_cast<T>(compute_inner_n_block(precision)));
+}
```

Comment:
The comment for the discussion:
I believe that `LDB` - is stride for second input of `Brgemm` block (not of `MatMul`).
1. If there is not `CopyB`, `LDB = snippets::utils::get_dim_stride(expr->get_input_port(1))`
2. If there is `CopyB`, we should take a stride as `CopyB` wrote. As far as I understand, `CopyB` stores the data in blocked layout. It means that `LDB` in this case should be aligned with `inner_block`: `rnd_up(N, inner_block)`. Since at the moment we use `max`, our LDB is not aligned with block size and we cannot support avx_vnni_2 in bf16 case (it [expects](https://github.com/openvinotoolkit/oneDNN/blob/6b99866a4531e38a74d1de36d5b366c54c5e6cc3/src/cpu/x64/brgemm/brgemm.cpp#L321) blocked weights).